subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncherFactory.java

```diff
@@ -140,6 +146,9 @@ private DefaultGradleLauncher doNewInstance(StartParameter startParameter, Gradl
         NestedBuildFactoryImpl nestedBuildFactory = new NestedBuildFactoryImpl(buildTreeScopeServices);
         serviceRegistry.add(NestedBuildFactory.class, nestedBuildFactory);
 
+        // Auto apply these requested plugins to the settings file
```

Comment:
This should live in the settings processor in some form, as this is it's job. This class shouldn't know about plugins.

Reply:
Yep